### ISSUE TITLE
Error: tensorflow/tensorflow/lite/java/ovic/demo/app/BUILD:9:1: no such package '@androidsdk//com.android.support': BUILD file not found in directory 'com.android.support' of external repository @androidsdk. Add a BUILD file to a directory to mark it as a package. and referenced by '//tensorflow/lite/java/ovic/demo/app:ovic_benchmarker_binary'

### ISSUE THREAD

**Issue by xinziXu** [tensorflow/tensorflow#38787]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- LG G8
- TensorFlow installed from (source or binary): binary
- TensorFlow version: 1.14.0
- Python version: 3.6.9
- Installed using virtualenv? pip? conda?: pip3
- Bazel version (if compiling from source): 
- GCC/Compiler version (if compiling from source): 8.3.0
- CUDA/cuDNN version: no
- GPU model and memory: no

**Describe the problem**
I trying to test the tflite model according to  [OVIC Benchmarker for LPCV 2020](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/java/ovic). 
I installed Bazel 3.0.0 using Bazel's apt repository.
I installed Android ndk17c.
I installed Android SDK using android-sdk_r24.4.1-linux.tgz by the sequent command:
`tar -zxvf android-sdk_r24.4.1-linux.tgz`
changed path for android SDK
used `android update sdk -u -a -t 8,213,58 ` to install SDK platform, build-tools and SDK API.
Then I use` ./configure` to config WORKSPACE for bazel.
My .tf_configure.bazelrc is like this:

> build --action_env PYTHON_BIN_PATH="/usr/bin/python3"
> build --action_env PYTHON_LIB_PATH="/usr/lib/python3/dist-packages"
> build --python_path="/usr/bin/python3"
> build --config=xla
> build:opt --copt=-march=native
> build:opt --copt=-Wno-sign-compare
> build:opt --host_copt=-march=native
> build:opt --define with_default_optimizations=true
> build --action_env ANDROID_NDK_HOME="/mnt/d/Share/software/android-ndk-r17c"
> build --action_env ANDROID_NDK_API_LEVEL="21"
> build --action_env ANDROID_BUILD_TOOLS_VERSION="28.0.3"
> build --action_env ANDROID_SDK_API_LEVEL="23"
> build --action_env ANDROID_SDK_HOME="/mnt/d/Share/software/android-sdk-linux"
> test --flaky_test_attempts=3
> test --test_size_filters=small,medium
> test:v1 --test_tag_filters=-benchmark-test,-no_oss,-gpu,-oss_serial
> test:v1 --build_tag_filters=-benchmark-test,-no_oss,-gpu
> test:v2 --test_tag_filters=-benchmark-test,-no_oss,-gpu,-oss_serial,-v1only
> test:v2 --build_tag_filters=-benchmark-test,-no_oss,-gpu,-v1only
> build --action_env TF_CONFIGURE_IOS="0"

Then I run
`bazel build -c opt --cxxopt=-Wno-all //tensorflow/lite/java/ovic/demo/app:ovic_benchmarker_binary`

Error occurred.

> ERROR: 
> /mnt/e/research/model_compression/proj/tensorflow/tensorflow/lite/java/ovic/demo/app/BUILD:9:1: no such package '@androidsdk//com.android.support': BUILD file not found in directory 'com.android.support' of external repository @androidsdk. Add a BUILD file to a directory to mark it as a package. and referenced by '//tensorflow/lite/java/ovic/demo/app:ovic_benchmarker_binary'
> ERROR: /mnt/e/research/model_compression/proj/tensorflow/tensorflow/lite/java/ovic/demo/app/BUILD:9:1: no such package '@androidsdk//com.android.support': BUILD file not found in directory 'com.android.support' of external repository @androidsdk. Add a BUILD file to a directory to mark it as a package. and referenced by '//tensorflow/lite/java/ovic/demo/app:ovic_benchmarker_binary'
> ERROR: Analysis of target '//tensorflow/lite/java/ovic/demo/app:ovic_benchmarker_binary' failed; build aborted: no such package '@androidsdk//com.android.support': BUILD file not found in directory 'com.android.support' of external repository @androidsdk. Add a BUILD file to a directory to mark it as a package.
> INFO: Elapsed time: 7.424s
> INFO: 0 processes.
> FAILED: Build did NOT complete successfully (18 packages loaded, 28 targets configured)

I have no idea how to fix this problem. And I tried to install Android Studio, it doesn't help either.

**Comment by A-suozhang** [comment]
I'm meeting the same issue with similar setting (instead i am using Ubuntu 18.04 but not windows), I manually installed the NDK r17c and install SDK via Android Studio.

**Comment by terryheo** [comment]
You need to run "./configure" first to configure Android NDK & SDK for build system.

During the configuration, you should answer "Yes" for the following question.
```
"Would you like to interactively configure ./WORKSPACE for Android builds?"
```
Then, you'll be asked for min Android NDK API level and Android SDK API level to use.

**Comment by A-suozhang** [comment]
> You need to run "./configure" first to configure Android NDK & SDK for build system.
> 
> During the configuration, you should answer "Yes" for the following question.
> 
> ```
> "Would you like to interactively configure ./WORKSPACE for Android builds?"
> ```
> 
> Then, you'll be asked for min Android NDK API level and Android SDK API level to use.

Of course I've done that... and only after doing that ```build --action_env ANDROID_NDK_HOME="/mnt/d/Share/software/android-ndk-r17c"``` could apppear in my .tf_configure.bazelrc

**Comment by terryheo** [comment]
Which branch did you use?
You said you used "TensorFlow version: 1.14.0" but it requires Bazel 0.26.2.

**Comment by A-suozhang** [comment]
> Which branch did you use?
> You said you used "TensorFlow version: 1.14.0" but it requires Bazel 0.26.2.

Sorry, my version differs a little with @xinziXu's, I used Tensorflow version:1.15.0 with bazel 0.25.1(within the requirement range in configure.py) on Ubuntu 18.04

**Comment by A-suozhang** [comment]
> Which branch did you use?
> You said you used "TensorFlow version: 1.14.0" but it requires Bazel 0.26.2.

Also, i'm a bit confused about when building the ovic example(running ```bazel build -c opt --cxxopt=-Wno-all //tensorflow/lite/java/ovic/demo/app:ovic_benchmarker_binary```), should i follow [this tutorial](https://s0www0tensorflow0org.icopy.site/lite/guide/android?hl=it) to build the tf-lite.aar and add it into the project via modifying the build.gradle(which iIdid and the build for the tf-lite.aar succeeded)

**Comment by terryheo** [comment]
Do you still have the issue?
If you do, could you share .tf_configure.bazelrc?

**Comment by dipendra009** [comment]
I am having the same issue but I am using Tensorflow 2.1.0 with Bazel 3.1.0. Do we have to add the tf-lite.aar file? I did build the TFLite using `bazel build -c opt --config=android_arm64 //tensorflow/lite:libtensorflowlite.so`. Could you please help me resolve it soon?

**Comment by terryheo** [comment]
@dipendra009 The ovic_benchmarker_binary should be built with bazel. You don't need to care of the library dependency since bazel handles it.

**Comment by dipendra009** [comment]
It is really weird, I have the android sdk and ndk installed and WORKSPACE configured, but it gives the same error: Projects/tensorflow/tensorflow/lite/java/ovic/demo/app/BUILD:9:1: no such package '@androidsdk//com.android.support': BUILD file not found in directory 'com.android.support' of external repository @androidsdk. Add a BUILD file to a directory to mark it as a package. and referenced by '//tensorflow/lite/java/ovic/demo/app:ovic_benchmarker_binary'

What would you suggest to resolve this error? I tried reinstalling multiple times, but I have no idea why it is giving the error.

**Comment by A-suozhang** [comment]
> It is really weird, I have the android sdk and ndk installed and WORKSPACE configured, but it gives the same error: Projects/tensorflow/tensorflow/lite/java/ovic/demo/app/BUILD:9:1: no such package '@androidsdk//com.android.support': BUILD file not found in directory 'com.android.support' of external repository @AndroidSDK. Add a BUILD file to a directory to mark it as a package. and referenced by '//tensorflow/lite/java/ovic/demo/app:ovic_benchmarker_binary'
> 
> What would you suggest to resolve this error? I tried reinstalling multiple times, but I have no idea why it is giving the error.

* After unbelievable trails and error, I found a weird solution, with sdk and ndk properly installed and added into tensorflow build .bazelrc,  just comment out the com.android.support in ovic apk BUILD, I could successfully build it this way, **However, no related tutorial or post mentions any related operation, so I'm not sure whether its the right way. (To be frank, its rather weird, even Im surprised it worked)*
* ![](https://github.com/A-suozhang/MyPicBed/raw/master//img/20200616183720.png)

**Comment by dipendra009** [comment]
Thank you, A-suozhang. That worked for me.

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.
